qcodes/instrument/ip_to_visa.py

```diff
@@ -90,6 +90,23 @@ def close(self) -> None:
         if getattr(self, 'visa_handle', None):
             self.visa_handle.close()
 
+        if getattr(self, "visabackend", None) == "sim" and getattr(
+            self, "resource_manager", None
+        ):
+            # The pyvisa-sim visalib has a session attribute but the resource manager is not generic in the
+            # visalib type so we cannot get it in a type safe way
+            known_sessions = getattr(self.resource_manager.visalib, "sessions", ())
+            session_found = self.resource_manager.session in known_sessions
+
+            n_sessions = len(known_sessions)
+            # if this instrument is the last one or there are no connected instruments its safe to reset the device
+            if (session_found and n_sessions == 1) or n_sessions == 0:
+                # work around for https://github.com/pyvisa/pyvisa-sim/issues/83
+                # see other issues for more context
+                # https://github.com/QCoDeS/Qcodes/issues/5356 and
+                # https://github.com/pyvisa/pyvisa-sim/issues/82
+                self.resource_manager.visalib._init()
```

Comment:
do we want to guard this with try/except for "_init" attribute defined in cacse pyvisa sim changes the implementation and/or name of this fucntion in the future? or is the damage very unlikely and only applicable to "sim" which is only used for tests and hence not in production, hence it's good as is?

Reply:
My original thought was that this is only used in testing but there is no harm in downgrading this to a warning if `_init` is not there.

Reply:
> downgrading this to a warning if _init is not there.

i like this!